usecases/backup/backupper_test.go

```diff
@@ -671,11 +673,50 @@ func TestManagerCoordinatedBackup(t *testing.T) {
 		assert.Contains(t, m.backupper.lastAsyncError.Error(), "abort")
 	})
 
+	t.Run("AbortCommit", func(t *testing.T) {
+		sourcer := &fakeSourcer{}
+		backend := &fakeBackend{}
+		m := createManager(sourcer, nil, backend, nil)
+
+		sourcer.On("Backupable", ctx, req.Classes).Return(nil)
+		ch := fakeBackupDescriptor(genClassDescriptions(cls, cls2)...)
+		sourcer.On("BackupDescriptors", any, backupID, mock.Anything).Return(ch).RunFn = func(a mock.Arguments) {
+			m.OnAbort(ctx, &AbortRequest{OpCreate, req.ID, backendName})
+		}
+		sourcer.On("ReleaseBackup", ctx, backupID, mock.Anything).Return(nil)
+
+		backend.On("HomeDir", mock.Anything).Return(path)
+		backend.On("GetObject", ctx, nodeHome, BackupFile).Return(nil, backup.NewErrNotFound(errors.New("not found")))
+		backend.On("Initialize", ctx, nodeHome).Return(nil)
+		backend.On("PutObject", mock.Anything, nodeHome, BackupFile, mock.Anything).Return(nil).Once()
+		backend.On("PutFile", mock.Anything, nodeHome, mock.Anything, mock.Anything).Return(nil)
+
+		req := req
+		req.Duration = time.Hour
+		got := m.OnCanCommit(ctx, &req)
+		want := &CanCommitResponse{OpCreate, req.ID, _TimeoutShardCommit, ""}
+		assert.Equal(t, got, want)
+
+		err := m.OnCommit(ctx, &StatusRequest{OpCreate, req.ID, backendName})
+		assert.Nil(t, err)
+		for i := 0; i < 20; i++ {
```

Comment:
Consider introducing a code seam (e.g., using interfaces, channels or other devices) to facilitate deterministic testing of async process.
- Apologies for the blatant suggestion if this is just wildly out of the question given the context. 🥹

Reply:
@noodnik2 Interesting! Do you mind elaborating more on how this might look?

Reply:
Certainly the test can be improved and will be tackled as part of upcoming refactoring tasks.
Thanks for your feedback